chromium_src/chrome/browser/resources/tab_search/auto_tab_groups/auto_tab_groups_page.ts

```diff
@@ -0,0 +1,264 @@
+      TabSearchApiProxyImpl.getInstance() as BraveTabSearchApiProxy
+
+  private visibilityChangedListener_: () => void
+  private elementVisibilityChangedListener_: IntersectionObserver
+  private isElementVisible_: boolean = false
+
+  static override get properties() {
+    return {
+      availableHeight: {type: Number},
+      showBackButton: {type: Boolean},
+      topic: {type: String},
+      topics_: {type: Array},
+      undoTopic_: {type: String},
+      isLoadingTopics: {type: Boolean},
+      errorMessage: {type: String},
+      needsPremium: {type: Boolean},
+    }
+  }
+
+  availableHeight: number = 0
+  showBackButton: boolean = false
+  isLoadingTopics: boolean = false
+  errorMessage: string = ''
+  needsPremium: boolean = false
+
+  static override get styles() {
+    return getCss()
+  }
+
+  constructor() {
+    super()
+
+    this.visibilityChangedListener_ = () => {
+      this.maybeUpdateSuggestedTopics_()
+    }
+
+    this.elementVisibilityChangedListener_ =
+        new IntersectionObserver((entries, _observer) => {
+          entries.forEach(entry => {
+            this.onElementVisibilityChanged_(entry.intersectionRatio > 0)
+          })
+        }, {root: document.documentElement})
+
+  }
+
+  protected onTopicInputChanged_(e: any) {
+    this.topic = e.value
+  }
+
+  protected onFocusTabsClicked_() {
+    this.getFocusTabs_(this.topic)
+  }
+
+  protected onUndoClicked_() {
+    this.apiProxy_.undoFocusTabs().then(() => {
+      this.undoTopic_ = ''
+    })
+  }
+
+  private getFocusTabs_(topic: string) {
+    this.undoTopic_ = ''
+    this.apiProxy_.getFocusTabs(topic).then(({windowCreated, error}) => {
+      if (error) {
+        this.errorMessage = error.message
+        if (error.rateLimitedInfo) {
+          this.needsPremium = !error.rateLimitedInfo.isPremium
+        }
+        return
+      }
+      this.errorMessage = ''
+      if (windowCreated) {
+        this.undoTopic_ = topic
+      }
+    })
+  }
+
+  private maybeUpdateSuggestedTopics_() {
+    // Don't update if it's already loading topics or when it's not visible.
+    if (this.isLoadingTopics ||
+        document.visibilityState !== 'visible' ||
+        !this.isElementVisible_) {
+      return
+    }
+
+    this.isLoadingTopics = true
+    this.apiProxy_.getSuggestedTopics().then(({topics, error}) => {
+      this.isLoadingTopics = false
+      if (error) {
+        this.errorMessage = error.message
+        if (error.rateLimitedInfo) {
+          this.needsPremium = !error.rateLimitedInfo.isPremium
+        }
+        return
+      }
+      this.onSuggestTopicsUpdated(topics)
+      this.errorMessage = ''
+    })
+  }
+
+  private onSuggestTopicsUpdated(topics: string[]) {
+    this.topics_ = topics
+  }
+
+  protected getTopicId_(index: number): string {
+    return 'topicId' + index
+  }
+
+  protected getTopicWithoutEmoji_(topic: string): string {
+    return topic.replace(/[^\x20-\x7F]/g, '').trimStart()
```

Comment:
Can't we not replace `{Emoji}` here too?

Reply:
@Douglashdaniel thoughts on this?

Reply:
I had first attempted this, but didn't get the desired results so had to switch to this syntax.

Reply:
sgtm